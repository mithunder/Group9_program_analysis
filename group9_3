module group9_3_constant_propagation :
       read unknown;
       zero  := unknown * 0;
       one   := zero + 1;
       two   := zero + one * one - zero + one;
       three := one + (two + one) * two - two * two;
       four  := three * two - three + one;
       if three > two -> six   := one + two + three
       [] three < two -> one   := three
       [] three = two -> three := two
       fi;
       seven := six + one;
       eight := four * two + (unknown - unknown);
       a := one;
       do a > two -> divbyzero := one / zero
       [] a = two -> a := a + 1
       od;
       
       if unknown = unknown -> skip
       [] one = two -> six := unknown + six
       [] two = three -> three := unknown + three
       [] (one = 1 & unknown > 0) -> {eight := unknown + eight; seven := unknown + three + four}
       fi;

       write one;
       write two;
       write three;
       write four;
       write six;
       write seven;
       write eight
end
