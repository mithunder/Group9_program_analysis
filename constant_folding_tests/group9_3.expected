module group9_3_constant_propagation:
read unknown;
zero := 0;
one := 1;
two := 2;
three := 3;
four := 4;
if
   true ->
      six := 6
   [] false ->
      one := 3
   [] false ->
      three := 2
fi;
seven := (6+one);
eight := 8;
a := one;
do
   (a>2) ->
      divbyzero := (one/0)
   [] (a=2) ->
      a := (a+1)
od;
if
   (unknown=unknown) ->
      skip
   [] (one=2) ->
      six := (unknown+six)
   [] (2=three) ->
      three := (unknown+three)
   [] ((one=1)&(unknown>0)) ->
      eight := (unknown+eight);
      seven := ((unknown+three)+4)
fi;
write (one);
write (2);
write (three);
write (4);
write (six);
write (seven);
write (eight)
end

